src/Components/allocations/Allocations.tsx

```diff
@@ -0,0 +1,206 @@
+import { Button, List, ListItem, Paper, TextField, Typography } from '@mui/material'
+import { Box } from '@mui/system';
+import React, { useEffect, useState } from 'react'
+import { loadDataFromFirebase, updateDataInFirebase, writeIntoDataBase } from '../../dataBaseUtils/readWrite'
+import ModalWindow from '../ModalWindow';
+import deleteAllocation from './helper_functions/deleteAllocation';
+
+interface editIntfs {
+  status: boolean;
+  allocationId: number | undefined;
+}
+
+interface AllocationIntfc {
+  name: string | undefined;
+  id: number | undefined;
+}
+
+export default function Allocations() {
+  const [allocations, setAllocations] = useState<any[]>([]);
+  const [reRender, setReRender] = useState(0);
+  const [edit, setEdit] = useState<editIntfs>({ status: false, allocationId: undefined });
+  useEffect(() => {
+    getAllocations(setAllocations);
+  }, [reRender]);
+
+  const deleteAllocation2 = (allocation: any): void => {
+    updateDataInFirebase(`allocations/${allocation.id}`)
+      .then(() => {
+        setReRender(prev => prev + 1);
+      });
+  }
+
+  const AllocationItemEditable = (props: any) => {
+    const allocation: AllocationIntfc = props.allocation;
+    const [allocationName, setAllocationName] = useState<string>(allocation?.name || "");
+    const handleAllocationNameChange = (event: any) => {
+      setAllocationName(event.target.value)
+    }
+    const handleAllocationNameSave = () => {
```

Comment:
need to change function to use data via args -to refactor it from the parent components scope